lib/src/view/tournament/tournament_list_screen.dart

```diff
@@ -0,0 +1,271 @@
+      AsyncData(:final value) => PlatformWidget(
+        androidBuilder: (_) => _AndroidBody(tournaments: value),
+        iosBuilder: (_) => _CupertinoBody(tournaments: value),
+      ),
+      AsyncError(:final error) => Center(child: Text('Could not load tournaments: $error')),
+      _ => PlatformScaffold(
+        appBarTitle: Text(context.l10n.tournaments),
+        body: const Center(child: CircularProgressIndicator()),
+      ),
+    };
+  }
+}
+
+class _AndroidBody extends StatefulWidget {
+  const _AndroidBody({required this.tournaments});
+
+  final TournamentLists tournaments;
+
+  @override
+  State<_AndroidBody> createState() => _AndroidBodyState();
+}
+
+enum _ViewMode {
+  completed,
+  ongoing,
+  upcoming;
+
+  String l10n(BuildContext context) {
+    switch (this) {
+      case _ViewMode.completed:
+        // TODO l10n
+        return 'Completed';
+      case _ViewMode.ongoing:
+        return context.l10n.broadcastOngoing;
+      case _ViewMode.upcoming:
+        return context.l10n.broadcastUpcoming;
+    }
+  }
+}
+
+class _AndroidBodyState extends State<_AndroidBody> with TickerProviderStateMixin {
+  late final TabController _tabController;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(initialIndex: 1, length: 3, vsync: this);
+  }
+
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(context.l10n.tournaments),
+        bottom: TabBar(
+          controller: _tabController,
+          tabs: _ViewMode.values.map((mode) => Tab(text: mode.l10n(context))).toList(),
+        ),
+      ),
+      body: TabBarView(
+        controller: _tabController,
+        children: <Widget>[
+          _TournamentListBody(tournaments: widget.tournaments.finished),
+          _TournamentListBody(tournaments: widget.tournaments.started),
+          _TournamentListBody(tournaments: widget.tournaments.created),
+        ],
+      ),
+    );
+  }
+}
+
+class _CupertinoBody extends StatefulWidget {
+  const _CupertinoBody({required this.tournaments});
+
+  final TournamentLists tournaments;
+
+  @override
+  _CupertinoBodyState createState() => _CupertinoBodyState();
+}
+
+class _CupertinoBodyState extends State<_CupertinoBody> {
+  _ViewMode _selectedSegment = _ViewMode.ongoing;
+
+  void setViewMode(_ViewMode mode) {
+    setState(() {
+      _selectedSegment = mode;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: const CupertinoNavigationBar(),
+      child: SafeArea(
+        child: Column(
+          mainAxisSize: MainAxisSize.max,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: [
+            Padding(
+              padding: Styles.bodyPadding,
+              child: CupertinoSlidingSegmentedControl<_ViewMode>(
```

Comment:
Now that the `CupertinoPageScaffold` accepts a `bottom` parameter for the app bar, we don't do that anymore, which simplifies greatly the code.

So you can refactor to use a single `PlatformScaffold` and a single `TabBarView` for both platforms. See broadcast round screen or create custom game screen for examples.

Reply:
Ah yes that makes the code much simpler indeed, thanks!